### PR TITLE
ref(replay): Convert the DOM tab to use *Frame types

### PR DESCRIPTION
**Pull request by ryan953**

### CODE REVIEW

File: static/app/utils/replays/frame.tsx [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    case 'resource.fetch':
+    case 'resource.xhr':
+    default:
+      return 'gray300';
+  }
+}
+
+export function getBreadcrumbType(
+  frame: BreadcrumbFrame | SpanFrame | ErrorFrame
+): BreadcrumbType {
```
Comment (billyvg) [inline static/app/utils/replays/frame.tsx:60]: Can we add a note here that this is currently only used to determine what icon to show (`<BreadcrumbIcon>`)?

File: static/app/utils/replays/hooks/useCrumbHandlers.tsx [hunk 2]
```diff
@@ -21,28 +23,33 @@ function useCrumbHandlers(startTimestampMs: number = 0) {
...
+              ? item.offsetMs
+              : relativeTimeInMs(item.timestamp ?? '', startTimestampMs)
+          );
         }
 
         if (item.data && typeof item.data === 'object' && 'nodeId' in item.data) {
           // XXX: Kind of hacky, but mouseLeave does not fire if you move from a
           // crumb to a tooltip
           clearAllHighlights();
+          // @ts-expect-error: Property 'label' does not exist on type
```
Comment (billyvg) [inline static/app/utils/replays/hooks/useCrumbHandlers.tsx:52]: Ah IIRC `label` is used in the LCP frame

File: static/app/utils/replays/hooks/useCrumbHandlers.tsx [hunk 3]
```diff
@@ -75,7 +82,15 @@ function useCrumbHandlers(startTimestampMs: number = 0) {
   );
 
   const handleClick = useCallback(
-    (crumb: Crumb | NetworkSpan) => {
+    (crumb: Crumb | NetworkSpan | BreadcrumbFrame | SpanFrame) => {
+      if ('offsetMs' in crumb) {
+        const frame = crumb; // Finding `offsetMs` means we have a frame, not a crumb or span
```
Comment (billyvg) [inline static/app/utils/replays/hooks/useCrumbHandlers.tsx:87]: When/what are crumbs?

**Review by billyvg — APPROVED** [review]

File: static/app/utils/replays/hooks/useCrumbHandlers.tsx (hunk 3 above)
Comment (ryan953) [reply static/app/utils/replays/hooks/useCrumbHandlers.tsx:87]: `Crumb` is the old type name, event & error breadcrumbs (not network data).

Once all the tabs are moved over the types of the args can be narrowed back down

**Pull request merged**